docs_src/microservices/application/AppServiceConfigurable.md

```diff
@@ -174,86 +214,83 @@ Please refer to the function's detailed documentation by clicking the function n
+- `TimeInterval` - Amount of time to batch before sending batched items to the next function in the pipeline. Used with  'bytime' and 'bytimecount' modes
 
 **Example**
 
 ```toml
     [Writable.Pipeline.Functions.BatchByCount]
       [Writable.Pipeline.Functions.BatchByCount.Parameters]
+      Mode = "bytimecount" # can be 'bycount', 'bytime' or 'bytimecount'
       BatchThreshold = "30"
+      TimeInterval = "60s"
 ```
 
-### [BatchByTime](../BuiltIn/#batch)
 
-**Parameters**
 
-- `TimeInterval` - Time duration to batch before sending batched items to the next function in the pipeline.
+!!! edgey "EdgeX 2.0"
+    For EdgeX 2.0 the `BatchByCount`, `BatchByCount`, and `BatchByTimeCount` configurable pipeline functions have been replaced by single `Batch` configurable pipeline function with additional `Mode` parameter.
 
-**Example**
+### [Compress](BuiltIn/#compressions)
 
-```toml
-    [Writable.Pipeline.Functions.BatchByTime]
-      [Writable.Pipeline.Functions.BatchByTime.Parameters]
-      TimeInterval = "60s"
-```
-### [BatchByTimeAndCount](../BuiltIn/#batch)
 **Parameters**
 
-- `BatchThreshold` - The number of items to batch before sending batched items to the next function in the pipeline.
-- `TimeInterval` - Time duration to batch before sending batched items to the next function in the pipeline.
+- `Algorithm ` - Compression algorithm to use.  Can be 'gzip' or 'zlib'
 
 **Example**
 
 ```toml
-    [Writable.Pipeline.Functions.BatchByTimeAndCount]
-      [Writable.Pipeline.Functions.BatchByTimeAndCount.Parameters]
-      BatchThreshold = "30"
-      TimeInterval = "60s"
+    [Writable.Pipeline.Functions.Compress]
+      [Writable.Pipeline.Functions.Compress.Parameters]
+      Algorithm = "gzip"
 ```
-### [CompressWithGZIP](../BuiltIn/#gzip)
-
-**Parameters**
 
-none
 
-**Example**
-
-```toml
-    [Writable.Pipeline.Functions.CompressWithGZIP]
-```
 
-### [CompressWithZLIB](../BuiltIn/#zlib)
+!!! edgey "EdgeX 2.0"
+    For EdgeX 2.0 the `CompressWithGZIP` and `CompressWithZLIB` configurable pipeline functions have been replaced by the single `Compress` configurable pipeline function with additional `Algorithm ` parameter.
 
+### [Encrypt](BuiltIn/#Encryption)
 **Parameters**
 
-none
+- `Key` -  (optional) Encryption key used for the encryption. Required if not using Secret Store for the encryption key data
+- `InitVector` - Initialization vector used for the encryption.
+- `SecretPath` - (optional) Path in the `Secret Store` where the encryption key is located. Required if Key not specified.
+- `SecretName` - (optional) name of the secret for the encryption key in the `Secret Store`.  Required if Key not specified.
 
-**Example**
+**Examples**
 
 ```toml
-    [Writable.Pipeline.Functions.CompressWithZLIB]
+	# Encrypt with key specified in configuation
+	[Writable.Pipeline.Functions.Encrypt]
```

Comment:
Is this extra space needed? Will it mess up the toml parser?

Reply:
Nope, but I did notice inconstancy in the indention between examples. Have up dated them all to be consistent.